src/main/java/com/efimchick/ifmo/web/servlets/Servlet1.java

```diff
@@ -0,0 +1,160 @@
+package com.efimchick.ifmo.web.servlets;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.util.*;
+import java.io.*;
+
+@WebServlet(
+        name = "Servlet1",
+        urlPatterns = {"/calc"}
+)
+public class Servlet1 extends HttpServlet {
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws IOException {
+
+        PrintWriter out = resp.getWriter();
+        String equation = req.getParameter("equation");
+        while (hasVars(equation)){
+            StringBuilder withValues = new StringBuilder();
+            Map<String, String[]> params = req.getParameterMap();
+            for (int i = 0; i < equation.length(); i++) {
+                char elem = equation.charAt(i);
+                if (Character.isLetter(elem)) {
+                    String result = params.get(Character.toString(elem))[0];
+                    withValues.append(result);
+                }
+                else {
+                    withValues.append(elem);
+                }
+            }
+            equation = withValues.toString();
+        }
+        ArrayDeque<String> rpn = shuntingYard(equation);
+        out.println(calculate(rpn));
+        out.flush();
+        out.close();
+    }
+
+    // https://en.wikipedia.org/wiki/Shunting-yard_algorithm#The_algorithm_in_detail
+    private ArrayDeque<String> shuntingYard(String infix) {
+        ArrayDeque<String> postfix = new ArrayDeque<>();
+        ArrayDeque<String> operStack = new ArrayDeque<>();
+        StringBuilder temp = new StringBuilder();
+        int lastOper = 0;
+        boolean negative = false;
+        char token;
+        String top;
+
+        for (int i = 0; i < infix.length(); i++) {
+            token = infix.charAt(i);
+            if (Character.isDigit(token)) {
+                temp.append(token);
+            }
+            if (!Character.isDigit(token) || i == infix.length() - 1) {
+                if (negative) {
+                    postfix.offerLast("-" + temp.toString());
+                    negative = false;
+                }
+                else {
+                    postfix.offerLast(temp.toString());
+                }
+                temp.delete(0, temp.length());
+            }
+
+            if (isOper(token)) {
+                if ((lastOper == i-1 || operStack.peekLast() == "(") && token == '-') {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use equals() to compare strings instead of '==' or '!='](https://app.codacy.com/manual/thejerome/IFMO_Servlets_20190915/pullRequest?prid=4387090)